src/bin/pg_autoctl/keeper_pg_init.c

```diff
@@ -546,6 +546,7 @@ wait_until_primary_is_ready(Keeper *keeper,
 			KeeperStateData *keeperState = &(keeper->state);
 			int timeoutMs = PG_AUTOCTL_KEEPER_SLEEP_TIME * 1000;
 
+			(void) pgsql_listen(&(monitor->notificationClient), &((char *){ 0 }));
```

Comment:
We need at least a comment that explains why it's okay to send an empty list of channels here, and then I would rather avoid this advanced notation and just use a `const char *channels[] = { 0 };` variable.

Reply:
Yeah, that's clearly a hack. I only added it as a quick way to open a connection as there was no proper interface available.
 Instead of hacking pgsql_listen(), how about  exposing a lower level interface like pgsql_open_connection()?

Reply:
Yeah we don't want to expose the lower-level interface. The whole point is that ensuring libpq-level clean-up and “lifetime management” is pretty hard, so we want to have that all sit in the same place and remain kind of opaque to the higher levels.

Also I'm not sure why we now have an explicit call to `pgsql_listen` that we didn't have before. Maybe that's where the new API could happen?